src/resolution/graph.rs

```diff
@@ -177,20 +180,20 @@ impl ResolvedNodeIds {
 
 /// A pointer to a specific node in a graph path. The underlying node id
 /// may change as peer dependencies are created.
-#[derive(Clone, Debug)]
-struct NodeIdRef(Rc<RefCell<NodeId>>);
+#[derive(Debug)]
+struct NodeIdRef(Cell<NodeId>);
```

Comment:
No need for this to be in an `Rc<RefCell` or be cloneable. This should be faster.